src/workers/parser.js

```diff
@@ -1,17 +1,39 @@
-import parser from '../parsers/python3.min.js';
 import { getRuleCount, rules } from '../util/rules.js';
+import { LOAD_PARSER } from '../actions/parser'
+import { PARSE_SNIPPET } from '../actions/app'
+
+let parser = null;
 
 const onError = (err) => {console.error(err);}
 
 self.onmessage = ({ data : action }) => {
-  const {snippet} = action.payload;
-  const AST = parser(snippet, onError);
-  const ruleCounts = {}; getRuleCount(AST, ruleCounts);
-  self.postMessage({
-    type: action.type,
-    payload: {
-      AST,
-      ruleCounts,
-    },
-  })
+  switch (action.type) {
+    case PARSE_SNIPPET: {
+      if (!parser) break;
+      const { snippet } = action.payload;
+      const AST = parser(snippet, onError);
+      const ruleCounts = {}; getRuleCount(AST, ruleCounts);
+      self.postMessage({
+        type: action.type,
+        payload: {
+          AST,
+          ruleCounts,
+        },
+      })
+      break;
+    }
+    case LOAD_PARSER: {
+      const parserURL  = action.payload;
+      importScripts(parserURL);
+      parser = Codesplain_parse_python3
```

Comment:
Where does `Codesplain_parse_python3` come from? I don't see where it was declared.